File: 백준/Silver/11723. 집합/집합.cc
#include <iostream>
using namespace std;

int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int m, s = 0;
    cin >> m;

    while (m--)
    {
        string op; cin >> op;

        if (op == "all")
            s = (1 << 21) - 1;
        else if (op == "empty")
            s = 0;
        else
        {
            int x;
            cin >> x;
            if (op == "add")
                s |= (1 << x);
            else if (op == "remove")
                s &= ~(1 << x);
            else if (op == "check")
                cout << ((s & (1 << x)) ? 1 : 0) << '\n';
            else if (op == "toggle")
                s ^= (1 << x);
        }
    }
}

File: 백준/Silver/11723. 집합/집합.py
import sys
input = sys.stdin.readline

def operation_A(S_set, q):
    if q == 'all':
        S_set.clear()
        S_set.update(range(1, 21))
    else:
        S_set.clear()

def operation_B(S_set, q, n):
    if q == 'add':
        S_set.add(n)
    elif q == 'remove':
        S_set.discard(n)
    elif q == 'check':
        sys.stdout.write('1\n' if n in S_set else '0\n')
    elif q == 'toggle':
        if n in S_set:
            S_set.remove(n)
        else:
            S_set.add(n)

S = set()
for _ in range(int(input())):
    command = input().strip()
    if command == "all" or command == "empty":
        operation_A(S, command)
    else:
        q, n = command.split()
        operation_B(S, q, int(n))